docs/cartesian-product.adoc

```diff
@@ -119,6 +121,52 @@ To demonstrate with a table:
 | 6th test  | 4                 | "B"
 |===
 
+=== `@CartesianEnumSource`
+
+Not only numbers, characters and string values can be provided to your test method, but also `Enum` constants.
+For this purpose, the test method can be annotated with `@CartesianEnumSource`.
+
+[source,java]
+----
+enum MyEnum {
+	ONE, TWO, THREE
+}
+
+enum AnotherEnum {
+	ALPHA, BETA, GAMMA, DELTA
+}
+
+@CartesianProductTest
+@CartesianEnumSource(MyEnum.class)
+@CartesianEnumSource(value = AnotherEnum.class, names = { "ALPHA", "DELTA" }, mode = Mode.EXCLUDE)
+void testEnumValues(MyEnum myEnum, AnotherEnum anotherEnum) {
+	// passing test code
+}
+----
+
+Like `@CartesianValueSource`, `@CartesianEnumSource` is an annotation that might look familiar - it mimics JUnit's `@EnumSource`, except `@CartesianEnumSource` is repeatable.
+It also does NOT work with `@ParameterizedTest`.
+https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests-sources-EnumSource::[You can read about `@EnumSource` here.]
+
+The test `testEnumValues` is executed exactly six times.
+The first parameter can have any of the three constants `ONE`, `TWO` or `THREE`.
+The second parameter can have any of the two constants `BETA` or `GAMMA` (note the `EXCLUDE` mode applied to the other two constants).
```

Comment:
This is unfortunately not enough. You have to talk about the configuration options in a bit more detail.

Reply:
Fair point, will do.

Reply:
Done! It is mostly a shameless copy of the JUnit docs.